File: BoSeonPark/그래프와 순회/24479.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10 ** 6) # 재귀허용 깊이 늘려주는 코드 *필수*
def dfs(v, g, r):
    global cnt
    
    v[r] = cnt
    for i in g[r]:
        if v[i] == 0:
            cnt += 1
            dfs(v, g, i)

n, m, r = map(int, input().split())

visited = [0 for _ in range(n + 1)]
graph = [[] for _ in range(n + 1)]
cnt = 1
for _ in range(m):
    s, e = map(int, input().split())
    graph[s].append(e)
    graph[e].append(s)

for i in range(1, n + 1):
    if len(graph[i]) >= 2:
        graph[i].sort()

dfs(visited, graph, r)

for i in range(1, n + 1):
    print(visited[i])

File: 김하영/Graphs and Traversals/24479.js
const fs = require("fs");

const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";

const input = fs.readFileSync(filePath).toString().trim().split("\n");

const [N, M, R] = input
  .shift()
  .split(" ")
  .map((v) => +v);

const edges = input.map((v) => v.split(" ").map((v) => +v));

const graph = new Map();
edges.forEach(([from, to]) => {
  if (graph.has(from)) {
    graph.get(from).push(to);
  } else {
    graph.set(from, [to]);
  }
  if (graph.has(to)) {
    graph.get(to).push(from);
  } else {
    graph.set(to, [from]);
  }
});
console.log(graph);

let count = 1;

const visited = new Array(N + 1).fill(false);
const answer = new Array(N).fill(0);

const dfs = (node) => {
  visited[node] = true;
  answer[node - 1] = count++;
  if (!graph.has(node)) {
    return;
  }
  const nodes = [...graph.get(node)].sort((a, b) => a - b);
  for (x of nodes) {
    if (!visited[x]) {
      dfs(x);
    }
  }
};

dfs(R);

console.log(answer.join(" "));
